src/System Application/App/Azure Storage Services Authorization/src/ReadySAS/StorServAuthReadySAS.Codeunit.al

```diff
@@ -51,6 +51,12 @@ codeunit 9088 "Stor. Serv. Auth. Ready SAS" implements "Storage Service Authoriz
             SharedAccessSignature := DelChr(SharedAccessSignature, '<', '?');
     end;
 
+    [NonDebuggable]
+    procedure SetSharedAccessSignature(NewSharedAccessSignature: SecretText)
+    begin
+        SetSharedAccessSignature(NewSharedAccessSignature.Unwrap())
```

Comment:
Do not unwrap it here. Unwraps should only be done in the location where the value is being used. Then the `[NonDebuggable]` can also be removed.

Add additional overloads etc, if necessary.

Reply:
I've reworked the global var to SecretText so the Unwrap() is is only done in procedures where its required.